signals/commands/things/src/main/java/org/eclipse/ditto/signals/commands/things/modify/ModifyThing.java

```diff
@@ -117,13 +118,16 @@ public static ModifyThing fromJson(final String jsonString, final DittoHeaders d
      */
     public static ModifyThing fromJson(final JsonObject jsonObject, final DittoHeaders dittoHeaders) {
         return new CommandJsonDeserializer<ModifyThing>(TYPE, jsonObject).deserialize(() -> {
-            final String thingId = jsonObject.getValueOrThrow(ThingModifyCommand.JsonFields.JSON_THING_ID);
-
+            final String thingId;
             final JsonObject thingJsonObject = jsonObject.getValueOrThrow(JSON_THING);
             final Thing extractedThing = ThingsModelFactory.newThing(thingJsonObject);
-
             final JsonObject initialPolicyObject = jsonObject.getValue(JSON_INITIAL_POLICY).orElse(null);
 
+            final Optional<String> optionalThingID = jsonObject.getValue(ThingModifyCommand.JsonFields.JSON_THING_ID);
+            thingId = optionalThingID.orElseGet(() -> extractedThing.getId().orElseThrow(() ->
```

Comment:
why not declare the variable in this line like `final String thingId = ...`?
Please change